### ISSUE TITLE
S3 Connector -- update metadata backend to mongodb

### ISSUE THREAD

**Issue by LaurenSpiegel** [filecoin-project/motion#108]
update motion docker compose to use mongo metadata backend and include mongo docker.

**Comment by xinaxu** [comment]
@masih do you know what is this about?

**Comment by LaurenSpiegel** [comment]
currently the docker compose is using just a file system backend for the metadata zenko stores. This is not meant for production. There is a mongo db option that should be used instead.

**Comment by LaurenSpiegel** [comment]
@elijaharita , this seems like a P0 for actually guiding people to use motion with s3. otherwise, it's just in a demo local storage mode. cc @davidd8

**Comment by elijaharita** [comment]
@LaurenSpiegel which Dockerfile is this configured in?

**Comment by LaurenSpiegel** [comment]
it should be here -- https://github.com/filecoin-project/motion-s3-connector/blob/main/docker-compose.yaml#L123

you need to set cloudserver to use mongo and make sure you are including a mongo instance

**Comment by masih** [comment]
@elijaharita [this](https://github.com/filecoin-project/motion-s3-connector/pull/7) is the branch I started and have not got a chance to complete. Please feel free to take over it. thank you!

**Comment by Angelo-gh3990** [comment]
Maybe use yugabyte if it need high availability and scalability?

**Comment by masih** [comment]
>Maybe use yugabyte if it needs high availability and scalability?

This stack lives entirely on the ISV side, not SP.
My sense would be to avoid involving yugabyte here because: 1) there is no evidence yet to suggest that mongodb doesn't cut it for ISVs who want S3, and 2) I do not believe Yugabyte as a backing store is supported by Zenko cloudserver, and I am hesitent to spend time implementing this backing store support in Zenko cloud server/ Zenko arsenal.

**Comment by Angelo-gh3990** [comment]
I agree . Maybe re-use the mongdb from singularly... extending it with a few tables or something

**Comment by gammazero** [comment]
@elijaharita See also, https://github.com/filecoin-project/motion-s3-connector/pull/11
